eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java

```diff
@@ -172,6 +172,14 @@ public boolean update(IUpdateContext context) {
                 for (ComponentReference componentReference : component.getReference()) {
                     if (reference.getName().equals(componentReference.getName())) {
                         promotedReferences.add(componentReference);
+
+                        // fix the promote issue if we can here for SWITCHYARD-2836
+                        if (reference.getPromote().isEmpty()) {
+                            if (reference.getInterface().eClass().getName().
```

Comment:
we are sure that reference.getInterface() is not null and componentReference.getInterface() either?